src/main/java/io/github/syst3ms/skriptparser/util/Date.java

```diff
@@ -0,0 +1,145 @@
+package io.github.syst3ms.skriptparser.util;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.text.SimpleDateFormat;
+import java.time.*;
+import java.util.Locale;
+import java.util.TimeZone;
+
+/**
+ * Represents a date.
+ * @author Njol
+ */
+public class Date implements Comparable<Date> {
```

Comment:
I don't like this nasty habit of Skript to name classes the same way as standard Java classes. This should be renamed to something like SkriptDate, and the fully-qualified references to Java's Date replaced by imports.